packages/scalecube-transport/src/Transport.js

```diff
@@ -2,23 +2,33 @@
 import { Observable } from 'rxjs';
 import { Transport as TransportInterface } from './api/Transport';
 import { errors } from './errors';
-import { TransportProvider } from './api/TransportProvider';
+import { TransportClientProvider } from './api/TransportClientProvider';
 import { TransportProviderConfig, TransportRequest } from './api/types';
 
 export class Transport implements TransportInterface {
-  _provider: any;
+  _clientProvider: any;
+  _serverProvider: any;
 
   constructor() {
-    this._provider = null;
+    this._clientProvider = null;
+    this._serverProvider = null;
     return this;
   }
 
-  setProvider(Provider: Class<TransportProvider>, TransportProviderConfig: TransportProviderConfig): Promise<void> {
+  setProvider(Provider: Class<TransportClientProvider>, TransportProviderConfig: TransportProviderConfig): Promise<void> {
```

Comment:
setProvider(provider: TransportClientProvider|TransportServerProvider): Promise<void>;
I changed the signature. I think constructed provider will be better than class and config